src/main/webapp/js/webrtc_adaptor.js

```diff
@@ -787,20 +787,27 @@ function WebRTCAdaptor(initialValues)
 			thiz.remotePeerConnection[streamId].ontrack = function(event) {
 				thiz.onTrack(event, closedStreamId);
 			}
-			
+
 			if (thiz.isPlayMode == false) {
 				//open data channel if it's not play mode,
 				const dataChannelOptions = {
 						ordered: true,
 				};
-				var dataChannel = thiz.remotePeerConnection[streamId].createDataChannel(streamId, dataChannelOptions);
-				thiz.initDataChannel(streamId, dataChannel);
+
+				if(forPublishing) {
+					var dataChannel = thiz.remotePeerConnection[streamId].createDataChannel(streamId, dataChannelOptions);
+					thiz.initDataChannel(streamId, dataChannel);
+			  }
+
+				thiz.remotePeerConnection[streamId].ondatachannel = function(ev) {
```

Comment:
It seems that in all cases this line is being called. I mean the same block also exist in the else block. It's like a code duplication?

Secondly,  ant media server does not create data channel for publishers so that I don't think we need this ondatachannel callback if playmode is false.

Reply:
> It's like a code duplication?

yeah it is a code dublication which I will fix.

> Secondly, ant media server does not create data channel for publishers so that I don't think we need this ondatachannel callback if playmode is false.

In conference call the playmode is false but we need ondatachannel callback for the streams which are published from the other participants of the conference. They write their messages using datachannel of the stream which they publish and we need to listen to their messages in their streams. Otherwise we miss these messages.